presto-main/src/main/java/com/facebook/presto/server/remotetask/ReactorNettyHttp2Client.java

```diff
@@ -0,0 +1,333 @@
+
+        // Create HTTP/2 client
+        this.httpClient = HttpClient.create(pool)
+                .protocol(HttpProtocol.H2, HttpProtocol.HTTP11)
+                .runOn(loopResources, true)
+                .http2Settings(settings -> settings.maxConcurrentStreams(config.getMaxStreamPerChannel()))
+                .secure(spec -> spec.sslContext(sslContext));
+    }
+
+    @Override
+    public <T, E extends Exception> T execute(Request request, ResponseHandler<T, E> responseHandler)
+            throws E
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    public <T, E extends Exception> HttpResponseFuture<T> executeAsync(Request airliftRequest, ResponseHandler<T, E> responseHandler)
+    {
+        SettableFuture<Object> listenableFuture = SettableFuture.create();
+
+        // Set the request headers
+        HttpClient client = this.httpClient.headers(hdr -> {
+            for (Map.Entry<String, String> entry : airliftRequest.getHeaders().entries()) {
+                hdr.set(entry.getKey(), entry.getValue());
+            }
+        });
+
+        URI uri = airliftRequest.getUri();
+        switch (airliftRequest.getMethod()) {
+            case "GET":
+                client.get()
+                        .uri(uri)
+                        .responseSingle((response, bytes) -> bytes.asInputStream().zipWith(Mono.just(response)))
+                        .subscribe(t -> onSuccess(responseHandler, t.getT1(), t.getT2(), listenableFuture), e -> onError(listenableFuture, e), () -> onComplete(listenableFuture));
+                break;
+            case "POST":
+                byte[] postBytes = ((StaticBodyGenerator) airliftRequest.getBodyGenerator()).getBody();
+                client.post()
+                        .uri(uri)
+                        .send(ByteBufFlux.fromInbound(Mono.just(postBytes)))
+                        .responseSingle((response, bytes) -> bytes.asInputStream().zipWith(Mono.just(response)))
+                        .subscribe(t -> onSuccess(responseHandler, t.getT1(), t.getT2(), listenableFuture), e -> onError(listenableFuture, e), () -> onComplete(listenableFuture));
+                break;
+            case "DELETE":
+                client.delete()
+                        .uri(uri)
+                        .responseSingle((response, bytes) -> bytes.asInputStream().zipWith(Mono.just(response)))
+                        .subscribe(t -> onSuccess(responseHandler, t.getT1(), t.getT2(), listenableFuture), e -> onError(listenableFuture, e), () -> onComplete(listenableFuture));
+                break;
+            default:
+                throw new UnsupportedOperationException("Unexpected request: " + airliftRequest);
+        }
+
+        return new HttpResponseFuture()
+        {
+            @Override
+            public boolean cancel(boolean mayInterruptIfRunning)
+            {
+                return listenableFuture.cancel(mayInterruptIfRunning);
```

Comment:
This should cancel the request. Could you please check how to do this properly in Reactive? Otherwise on timeout a request may not get cancelled and it will lead to accumulation of requests.